components/new-contribution-flow/StepPayment.js

```diff
@@ -1,16 +1,283 @@
+      setNoPaymentMethods(true);
+      setLoadingSelectedPaymentMethod(false);
+    }
+  }, [paymentOptions]);
+
+  useEffect(() => {
+    onChange({ stepPayment: paymentObject });
+  }, [paymentObject]);
+
+  return (
+    <Box width={1}>
+      {LoggedInUser ? (
+        <div>
+          {loadingSelectedPaymentMethod ? (
+            <Loading />
+          ) : noPaymentMethods ? (
+            <MessageBox type="warning" withIcon>
+              <FormattedMessage
+                id="NewContribute.noPaymentMethod"
+                defaultMessage="There are no payment methods available."
+              />
+            </MessageBox>
+          ) : (
+            <StyledRadioList
+              id="PaymentMethod"
+              name="PaymentMethod"
+              keyGetter="key"
+              options={paymentOptions}
+              onChange={setNewPaymentMethod}
+              value={stepPayment?.key}
+            >
+              {({ radio, checked, value: { key, title, subtitle, icon } }) => (
+                <PaymentMethodBox minheight={50} py={2} bg="white.full" data-cy="recurring-contribution-pm-box" px={3}>
+                  <Flex alignItems="center">
+                    <Box as="span" mr={3} flexWrap="wrap">
+                      {radio}
+                    </Box>
+                    <Flex mr={2} css={{ flexBasis: '26px' }}>
+                      {icon}
+                    </Flex>
+                    <Flex flexDirection="column">
+                      <P fontSize="12px" fontWeight={subtitle ? 600 : 400} color="black.900">
+                        {title}
+                      </P>
+                      {subtitle && (
+                        <P fontSize="Caption" fontWeight={400} lineHeight="Caption" color="black.500">
+                          {subtitle}
+                        </P>
+                      )}
+                    </Flex>
+                  </Flex>
+                  {key === 'newCreditCard' && checked && (
+                    <Box my={3}>
+                      <NewCreditCardForm
+                        name="newCreditCardInfo"
+                        // profileType={get(this.props.collective, 'type')}
+                        // error={errors.newCreditCardInfo}
+                        profileType={'USER'}
+                        onChange={setNewPaymentMethod}
+                        onReady={({ stripe }) => setStripe(stripe)}
+                        // hidePostalCode={this.props.hideCreditCardPostalCode}
+                      />
+                    </Box>
+                  )}
+                  {key === 'manual' && checked && withManual?.instructions && (
+                    <Box my={3} color="black.600" fontSize="Paragraph">
+                      {withManual?.instructions}
+                    </Box>
+                  )}
+                </PaymentMethodBox>
+              )}
+            </StyledRadioList>
+          )}
+        </div>
+      ) : (
+        <div>not logged in</div>
```

Comment:
What's planned there?

Reply:
Had not got to it yet. I imagine it'll be relatively similar to the logged in flow, just not having saved credit cards, etc